analysis/src/TypeUtils.ml

```diff
@@ -48,6 +112,59 @@ let instantiateType ~typeParams ~typeArgs (t : Types.type_expr) =
     in
     loop t
 
+let instantiateType2 ?(typeArgContext : typeArgContext option)
```

Comment:
Slightly different to the existing `instantiateType` - this instantiates via comparing `Tvar` name only, while the existing one structurally checks whether the `Tvar` types are the same. In this new solution they won't be the exact same object since we're moving across type manifests and what not.